src/Illuminate/Collections/LazyCollection.php

```diff
@@ -843,6 +843,20 @@ public function mapWithKeys(callable $callback)
         });
     }
 
+    /**
+     * Run an associative map over each of the item's values filtered by their key.
+     *
+     * The callback should return the same resultset with the modified values of the given keys.
+     *
+     * @param  string|array  $keys
+     * @param  callable  $callback
+     * @return static
+     */
+    public function mapByKey($keys, callable $callback)
+    {
+        return $this->passthru('mapByKey', func_get_args());
```

Comment:
I'm not 100% sure this is the correct way to go about adding a method to LazyCollection.